bokehjs/src/lib/core/has_props.ts

```diff
@@ -468,7 +468,7 @@ export abstract class HasProps extends Signalable() implements Equatable, Printa
     return changed
   }
 
-  setv(changed_attrs: Attrs, options: HasProps.SetOptions = {}): void {
+  setv<T extends Attrs>(changed_attrs: Partial<T>, options: HasProps.SetOptions = {}): void {
```

Comment:
In theory it should be possible to infer `T` from `typeof this["properties"]`, but that works in some contexts, but not in all. When it doesn't work, then `T == any` and we get a dynamically typed API with no warning.

Reply:
I think this would be an unacceptable burden if end-users were required to specify the type but as `setv` is really internal API, practically speaking, we can just deal with it.